### ISSUE TITLE
Add wake oscillator feature 

### ISSUE THREAD

**Issue by mvanzulli** [ONSAS/ONSAS#468]
This enhancement issue seek to generalize the ONSAS implementation in order to include the Wake Oscillator Model as an improvement of the aerodynamic forces.

 - [x] add VIV function
 - [x] change authorship list

**Comment by mvanzulli** [comment]
What about the `boundaryConds.loadsCoordSys`field which refers to `global` or `local` coordinantes

**Comment by jorgepz** [comment]
I would integrate that as a cell together with the forcescase vector....

**Comment by mvanzulli** [comment]
The new general ONSAS line call should be 
```
[matUs,matTemp, matQ, loadFactors, uControlDofCase1,  TControlDofCase1,  qControlDofCase1] = ONSAS( materials, elementSettings, elementsInfo, boundaryConds, initialConds, nodesCoords, analysisSettings, otherParams ) ;
```  ? 

Maybe we should try to compact that ? 
output = ONSAS( materials, elements, boundaryConds, initialConds, nodesCoords, analysisSettings, otherParams ) ;

and output is a struct with ´output.matUs = matUs´ and so on. Also `elements.elementsInfo` and `elements.elementSettings`.

**Comment by jorgepz** [comment]
Not to be implemented yet.

**Comment by jorgepz** [comment]
I'll reopen this issue with a different approach. just including the WOM computation as an additional feature but not as monolithic by the moment.

**Comment by jorgepz** [comment]
advances done in #504

**Comment by jorgepz** [comment]
the VIV function was added by @TheChikenfly in https://github.com/ONSAS/ONSAS.m/pull/518
 we have yet to update the authorship list before closing this issue.

**Comment by TheChikenfly** [comment]
Amazing thanks!

**Comment by jorgepz** [comment]
We just fixed/optimized ONSAS assembler flow. and also updated the authors list.
To this issue description I just added  "change global VIV variables to official fields in the input structs" . we can discuss this later.

**Comment by jorgepz** [comment]
we discuss with @TheChikenfly  and @mvanzulli  and decide to leave qvect as global variable by the moment

**Comment by mvanzulli** [comment]
I think we need to add a simple problem into tests. I've been changing some files and the VIV feature is not being tested

**Comment by jorgepz** [comment]
Indeed! That is very important.

**Comment by jorgepz** [comment]
I added a subtask to improve the VIV feature considering https://github.com/ONSAS/ONSAS.m/pull/596

**Comment by jorgepz** [comment]
@mvanzulli @TheChikenfly I think it would be nice to modify this example since the verification solution is a numeric matrix with no relation to the solution of a coupled differential equation system as for instance, in Example 1 from the JFS paper. I suggest (and I'll set as a someday change) remove this example and add wom and womIL verifications in the dragreconfiguration example, adding a dynamic case analysis after the static ones. What do you think?

**Comment by mvanzulli** [comment]
> I think it would be nice to modify this example since the verification solution is a numeric matrix with no relation to the solution of a coupled differential equation system as for instance, in Example 1 from the JFS paper.

Right, so might be better to implement the validation exactly as it is in the first example of the paper and derive the numeric solution from the  equations:

![Image](https://github.com/user-attachments/assets/da162f65-a11a-4976-bbed-e634027edf44)

is that what you mean ? 

> suggest (and I'll set as a someday change) remove this example and add wom and womIL verifications in the dragreconfiguration example

Well I think it is better to have it separately as it is a different feature and the reconfiguration example has enough content, so I would prefer to create a different example section, but we can add a link between them.

**Comment by jorgepz** [comment]
keep in mind that the code for the jfs paper is published and the example 1 in particular is in https://github.com/TheChikenfly/source_code_VIVpaper/tree/main/example_1

**Comment by mvanzulli** [comment]
I've been reviewing the current global variables we have:

- `constantLiftDir` and `uniformUdot` were added for validation purposes, as some literature assumes a constant lift direction for computing the lift force. However, in the formulation from [[this paper](https://scholar.google.com/citations?view_op=view_citation&hl=pt-BR&user=mwDZNb4AAAAJ&citation_for_view=mwDZNb4AAAAJ:2osOgNQ5qMEC)](https://scholar.google.com/citations?view_op=view_citation&hl=pt-BR&user=mwDZNb4AAAAJ&citation_for_view=mwDZNb4AAAAJ:2osOgNQ5qMEC) (see line 154), this assumption is not made.  
The difference is that one assumes a constant lift direction and uniform velocity field, while the other considers it constant only as an average at the nodes.

As far as I remember, these were only implemented to validate against the literature. **I propose we remove them**, since our implementation and examples don’t rely on these assumptions.

The other variables — `ILVIVBool`, `addedMassBool`, and `VIVBool` — make sense to keep. We could also consider setting `ILVIVBool = true` whenever `VIVBool` is true, to always include the in-line WOM oscillator along with the cross-flow. However, since this is a second-order effect, having it as an optional flag in the `analysisSettings` struct might be more appropriate.

So I will start adding a case in the drag reconfiguration considering VIV and after that if everything works we can clean these globals.

**Comment by jorgepz** [comment]
thanks for reviewing that @mvanzulli ! Perfect, I agree on removing both globals.

In ILVIVBool I suggest setting it as false by default and keep them separated (since, I think, it does not represent any additional code refactoring).

**Comment by mvanzulli** [comment]
Makes sense, will then assign more informative names of these parameters `cross_flow_viv` and `in_line_viv`. Also, this feature requires adding as global the q matrix, but initialization can be infered from the analysis so one option is to do it in:

```
  %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
  % TEMPORARY
  %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
  global spitMatrices
  if spitMatrices == true
    save('-mat', 'output/loads.mat', 'factorLoadsFextCell');
  end

 if corss_flow_viv 
   global qvect;
    qvect = zeros(numElements * 2, numTimeSteps);
    qvect(1:2:end, 1) = (2 * rand(numElements, 1) - 1) * 0.01;
  end
``` 

ideally we would like to have it in the dofs, but that is a larger refactor, maybe this intermediate step is an option.

**Comment by mvanzulli** [comment]
I generted the following simulation form an intial state of the deformed configuration, do you think is okay ? There some vibrations and oscilations in and also osilation arround the equilibrium position as now we need to include a lift force: 

https://github.com/user-attachments/assets/f444a34c-574f-4897-85ad-7d56e775edc7
